Extensions/RosettaTorch/Includes/AlphaZero/SelfPlay/Runner.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019 Chris Ohk, Youngjoong Kim, SeungHyun Jeon
+
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+// It is based on peter1591's hearthstone-ai repository.
+// References: https://github.com/peter1591/hearthstone-ai
+
+#ifndef ROSETTASTONE_TORCH_ALPHA_ZERO_SELF_PLAY_RUNNER_HPP
+#define ROSETTASTONE_TORCH_ALPHA_ZERO_SELF_PLAY_RUNNER_HPP
+
+#include <AlphaZero/Logger/ILogger.hpp>
+#include <AlphaZero/SelfPlay/RunOptions.hpp>
+#include <AlphaZero/SelfPlay/SelfPlayer.hpp>
+#include <AlphaZero/Utils/ThreadRunner.hpp>
+
+namespace RosettaTorch::AlphaZero::SelfPlay
+{
+//!
+//! \brief Runner class.
+//!
+//! This class is a simple runner for self player.
+//!
+class Runner
+{
+ public:
+    //! Constructs runner with given \p logger.
+    //! \param logger The logger to write the log.
+    Runner(ILogger& logger);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'Runner' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/utilForever/RosettaStone/pullRequest?prid=4565217)